builtin/providers/aws/resource_aws_api_gateway_client_certificate.go

```diff
@@ -0,0 +1,129 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayClientCertificate() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayClientCertificateCreate,
+		Read:   resourceAwsApiGatewayClientCertificateRead,
+		Update: resourceAwsApiGatewayClientCertificateUpdate,
+		Delete: resourceAwsApiGatewayClientCertificateDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"created_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"expiration_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"pem_encoded_certificate": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayClientCertificateCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	input := apigateway.GenerateClientCertificateInput{}
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+	log.Printf("[DEBUG] Generating API Gateway Client Certificate: %s", input)
+	out, err := conn.GenerateClientCertificate(&input)
+	if err != nil {
+		return fmt.Errorf("Failed to generate client certificate: %s", err)
+	}
+
+	d.SetId(*out.ClientCertificateId)
+	d.Set("description", out.Description)
```

Comment:
It can, I just thought we could save an API call, but I totally don't mind removing these `Set`s from Create. I know it was kind of convention in most other resources.

I think it would be worth documenting these conventions somewhere, so we can point people there and be consistent when demanding changes in other people's PRs. 😃 

Happy to discuss this on Slack.